src/goto-symex/symex_main.cpp

```diff
@@ -242,12 +242,31 @@ void goto_symext::symex_with_state(
   const get_goto_functiont &get_goto_function,
   symbol_tablet &new_symbol_table)
 {
+  struct reset_namespacet
```

Comment:
Maybe add a one-liner explaining what this aims to achieve.

Reply:
Done!